project-koku/koku
# [COST-3315] partition gcp openshift daily

## Jira Ticket

[COST-3315](https://issues.redhat.com/browse/COST-3315)

## Description

This change will adjust our OCP on GCP data to be partitioned daily as well. SQL updates to match those changes. It also adjusts get_openshift_topology_for_multiple_providers to return a list of topologies that each pertain to a specific clusters information and updates to the match OpenShift resources functions to match those changes.

## Testing

1. Checkout Branch
2. Restart Koku
3. Load test customer data: `make create_test_customer` followed by `make load_test_customer_data`
4. Verify that the OCP on GCP data has loaded: http://localhost:8000/api/cost-management/v1/reports/openshift/infrastructures/gcp/costs/
5. Log into trino using the trino CLI: `trino --server localhost:8080 --catalog hive --schema org1234567 --user admin --debug`
6. Verify that the gcp_openshift_daily is now partitioned by day as well: `show create table gcp_openshift_daily;' You should see `partitioned_by = ARRAY['source','year','month','day'] ` at the end
7. smoke tests should cover calculation checks.


## Testing the transition from main to this branch:

1. checkout main
2. restart koku
3. in a local IQE setup, put a pdb at the top of the `test_api_ocp_on_gcp_source_raw_calc` test
4. run the raw calc test until it stops at the pdb
5. checkout this branch
6. run the migrate trino script: `python scripts/cji_scripts/migrate_trino.py`
7. hit the process ocp on cloud endpoint with the relevant provider, example call: 
http://localhost:5042/api/cost-management/v1/report/process/openshift_on_cloud/?provider_uuid=a3fbdeb3-d78c-4e4e-b6c4-c534ff0e52fb&schema=org1234567&start_date=2023-01-01

8. resummarize that provider for the last two months, example call:
http://localhost:5042/api/cost-management/v1/report_data/?schema=org1234567&provider_type=GCP-local&provider_uuid=ea684142-f2a2-48eb-b635-425060553fb0&start_date=2023-01-01

http://localhost:5042/api/cost-management/v1/report_data/?schema=org1234567&provider_type=GCP-local&provider_uuid=ea684142-f2a2-48eb-b635-425060553fb0&start_date=2023-02-01

9. continue with the smoke test and it should pass
## Notes

...
